### PR TITLE
Feature/angular velocity updated

### PR DESCRIPTION
**Pull request by emielsteerneman**
Code Review time

### CODE REVIEW

File: python_utils/testRobot.py [hunk 1]
```diff
@@ -17,17 +17,20 @@
...
-from roboteam_embedded_messages.python.REM_RobotPIDGains import REM_RobotPIDGains
+from roboteam_embedded_messages.python.REM_RobotSetPIDGains import REM_RobotSetPIDGains
 from roboteam_embedded_messages.python.REM_RobotPIDGains import REM_RobotPIDGains
 from roboteam_embedded_messages.python.REM_RobotLog import REM_RobotLog
 from roboteam_embedded_messages.python.REM_BasestationLog import REM_BasestationLog
 from roboteam_embedded_messages.python.REM_BasestationGetConfiguration import REM_BasestationGetConfiguration
 from roboteam_embedded_messages.python.REM_BasestationSetConfiguration import REM_BasestationSetConfiguration
 from roboteam_embedded_messages.python.REM_BasestationConfiguration import REM_BasestationConfiguration
 
+robotStateInfoFile = open(f"PIDfiles/robotStateInfo_{int(time.time())}.csv", "w")
```
Comment (emielsteerneman) [inline python_utils/testRobot.py]: Please rename 'PIDfiles' to something like 'logs'

File: python_utils/testRobot.py [hunk 2]
```diff
@@ -107,6 +108,38 @@ def normalize_angle(angle):
 # stlink_port = "/dev/serial/by-id/usb-STMicroelectronics_STM32_STLink_0674FF525750877267181714-if02"
 stlink_port = "/dev/serial/by-id/usb-STMicroelectronics_STM32_STLink_066FFF544852707267223637-if02"
 
+def createSetPIDCommand(PbodyX = 0.0, IbodyX = 0.0, DbodyX = 0.0, PbodyY = 0.0, IbodyY = 0.0, DbodyY = 0.0, PbodyW = 0.0, IbodyW = 0.0, DbodyW = 0.0, PbodyYaw = 0.0, IbodyYaw = 0.0, DbodyYaw = 0.0, Pwheels = 0.0, Iwheels = 0.0, Dwheels = 0.0):
```
Comment (emielsteerneman) [inline python_utils/testRobot.py]: Add robot_id as argument to function

File: python_utils/testRobot.py [hunk 3]
```diff
@@ -146,29 +180,40 @@ def createRobotCommand(robot_id, test, tick_counter, period_fraction):
 		log = "speed = %.2f" % cmd.dribbler
 
 	if test == "rotate":
+		cmd.angularControl = 1
```
Comment (emielsteerneman) [inline python_utils/testRobot.py]: I think we should rename this variable to something like `useAbsoluteAngle` or something. It should be immediately clear what it means if this variable is true or false

File: python_utils/testRobot.py [hunk 4]
```diff
@@ -241,6 +286,27 @@ def createRobotCommand(robot_id, test, tick_counter, period_fraction):
 				basestation.write(cmd_encoded)
 				parser.writeBytes(cmd_encoded)
 				last_robotcommand_time = time.time()
+				
+				# Create and send new PID gains (default 0)
+				setPID = createSetPIDCommand() #put PID gains as arguments to change them
+				setPID_encoded = setPID.encode()
+				basestation.write(setPID_encoded)
```
Comment (emielsteerneman) [inline python_utils/testRobot.py]: setPIDGains should not be written 60 times per second. Either once, or maybe at most once per second

**Review by emielsteerneman — COMMENTED** [review]
Some small stuff but lgtm;

**Comment by skaeringur97** [comment]
Tested it and it works. Can be merged on Wednesday if there are no comments

**Review by robopc — APPROVED** [review]
LGTM

**Pull request merged**